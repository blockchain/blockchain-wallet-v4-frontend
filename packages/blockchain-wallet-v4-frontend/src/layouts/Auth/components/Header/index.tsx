import React from 'react'
import styled from 'styled-components'

import { Image, Link } from 'blockchain-info-components'
import { ServiceAnnouncement } from 'components/Announcements'
import { Brand, Public } from 'components/Navbar'
import { isMobile, media } from 'services/styles'

const qsParams = new URLSearchParams(window.location.hash)
const isLatam = qsParams.has('latam')
const isSofi = window.location.hash.includes('sofi')

const NavbarStyled = styled(Public)<{ authProduct: string }>`
  padding: 0 16px;
  box-sizing: border-box;
  background-color: ${(props) =>
    props.authProduct === 'EXCHANGE'
      ? props.theme.exchangeLogin
      : isLatam
      ? '#04001F'
      : props.theme.grey900};
  background-image: ${(props) =>
    props.authProduct !== 'EXCHANGE' && isLatam
      ? `url('/img/bg-pattern-latam.svg')`
      : `url('/img/bg-pattern.svg')`};
`
const NavbarBrandStyled = styled(Brand)`
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: center;
  padding-left: 0;
`
const BlockchainLogoImage = styled(Image)`
  width: 240px;
  display: block;
  ${media.tablet`
    width: 200px;
  `}
`
const BlockchainSofiLogoImage = styled(Image)`
  width: 375 px;
  display: block;
  ${media.tablet`
    width: 275px;
  `}
`
const AuthBrand = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`
const HeaderLink = styled(Link)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`

const Header = (props) => {
  const nabvarHeight = isSofi && isMobile() ? '55px' : '112px'
  return (
    <>
      <NavbarStyled height={nabvarHeight} authProduct={props.authProduct}>
        <NavbarBrandStyled>
          <AuthBrand>
            <HeaderLink href='https://www.blockchain.com'>
              {isLatam && <BlockchainLogoImage name='sesocio-to-blockchain-logo' height='70px' />}{' '}
              {isSofi && <BlockchainSofiLogoImage name='blockchain-sofi-logo' height='24px' />}
              {!isLatam && !isSofi && <BlockchainLogoImage name='blockchain-logo' height='24px' />}
            </HeaderLink>
          </AuthBrand>
        </NavbarBrandStyled>
      </NavbarStyled>
      <ServiceAnnouncement alertArea='public' />
    </>
  )
}

export default Header
