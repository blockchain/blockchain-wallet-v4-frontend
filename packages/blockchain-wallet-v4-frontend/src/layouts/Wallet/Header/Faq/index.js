import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link, Icon } from 'blockchain-info-components'

const FaqLink = styled(Link)`
  position: relative;
  margin-top: 4px;
  
  & > :first-child:hover { 
    cursor: pointer;
  }
  
  ::after {
    opacity: ${props => props.trayRightOpen ? '1' : '0'};
    content: "";
    position: absolute;
    top: 24px;
    left: -2px;
    width: 0;
    height: 0;
    z-index: 3;
    border-left: 11px solid transparent;
    border-right: 11px solid transparent;
    border-bottom: 16px solid ${props => props.theme['white-blue']};
    transition: opacity ${props => props.trayRightOpen ? '.2s' : '0'};
    transition-delay: ${props => props.trayRightOpen ? '.3s' : '.1s'};
  }
`

const Faq = (props) => {
  return (
    <FaqLink {...props} size='16px' weight={300} color='white'>
      <Icon name='question-in-circle-filled' size='18px' color='white'/>
    </FaqLink>
  )
}

Faq.propTypes = {
  onClick: PropTypes.func.isRequired,
  trayRightOpen: PropTypes.bool.isRequired
}

export default Faq
