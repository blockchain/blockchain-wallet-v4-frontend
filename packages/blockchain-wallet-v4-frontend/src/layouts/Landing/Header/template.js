import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { NavLink } from 'react-router-dom'

import { Button, Image, Link } from 'blockchain-info-components'
import { Navbar, NavbarBrand, NavbarHeader, NavbarMenu, NavbarNav, NavbarNavItem, NavbarToggler } from 'components/Navbar'

const ButtonNav = styled(NavbarNav)`
  flex-direction: row;
  width: 340px;
  margin: 0 auto;
  @media(min-width: 768px) and (max-width: 992px) { display: none; }
`

const Header = (props) => {
  const { toggled, handleToggle } = props

  return (
    <Navbar height='90px' fluid>
      <NavbarHeader>
        <NavbarBrand>
          <NavLink to='/'>
            <Image name='blockchain-vector' height='20px' />
          </NavLink>
        </NavbarBrand>
      </NavbarHeader>
      <NavbarMenu toggled={toggled}>
        <NavbarNav>
          <NavbarNavItem>
            <NavLink to='/wallet'>
              <FormattedMessage id='layouts.landing.header.wallets' defaultMessage='Wallets' />
            </NavLink>
          </NavbarNavItem>
          <NavbarNavItem>
            <Link href='https://blockchain.info/charts' target='_blank'>
              <FormattedMessage id='layouts.landing.header.charts' defaultMessage='Charts' />
            </Link>
          </NavbarNavItem>
          <NavbarNavItem>
            <Link href='https://blockchain.info/stats' target='_blank'>
              <FormattedMessage id='layouts.landing.header.stats' defaultMessage='Stats' />
            </Link>
          </NavbarNavItem>
          <NavbarNavItem>
            <Link href='https://blockchain.info/markets' target='_blank'>
              <FormattedMessage id='layouts.landing.header.markets' defaultMessage='Markets' />
            </Link>
          </NavbarNavItem>
          <NavbarNavItem>
            <Link href='https://blockchain.info/api' target='_blank'>
              <FormattedMessage id='layouts.landing.header.api' defaultMessage='Api' />
            </Link>
          </NavbarNavItem>
        </NavbarNav>
        <ButtonNav>
          <NavbarNavItem>
            <NavLink to='/login'>
              <Button nature='secondary' uppercase rounded>
                <FormattedMessage id='layouts.landing.header.login' defaultMessage='Log in' />
              </Button>
            </NavLink>
          </NavbarNavItem>
          <NavbarNavItem>
            <NavLink to='/register'>
              <Button nature='primary' uppercase rounded>
                <FormattedMessage id='layouts.landing.header.signup' defaultMessage='Sign up' />
              </Button>
            </NavLink>
          </NavbarNavItem>
        </ButtonNav>
      </NavbarMenu>
      <NavbarToggler onToggle={handleToggle} />
    </Navbar>
  )
}

Header.propTypes = {
  handleToggle: PropTypes.func.isRequired,
  toggled: PropTypes.bool.isRequired
}

export default Header
