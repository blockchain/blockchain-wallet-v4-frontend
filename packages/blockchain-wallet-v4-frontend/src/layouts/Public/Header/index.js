import React from 'react'

import Announcements from 'components/Announcements'
import { Image, Link } from 'blockchain-info-components'
import { Navbar, NavbarBrand } from 'components/Navbar'

const Header = () => {
  return (
    <React.Fragment>
      <Navbar height='90px'>
        <NavbarBrand>
          <Link href='https://www.blockchain.com'>
            <Image name='blockchain-vector' height='20px' />
          </Link>
        </NavbarBrand>
      </Navbar>
      <Announcements type='service' alertArea='public' />
    </React.Fragment>
  )
}

export default Header
