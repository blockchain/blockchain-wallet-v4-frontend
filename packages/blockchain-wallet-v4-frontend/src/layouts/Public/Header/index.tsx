import React from 'react'

import { Image, Link } from 'blockchain-info-components'
import { Navbar, NavbarBrand } from 'components/Navbar'
import Announcements from 'components/Announcements'
import media from 'services/ResponsiveService'
import styled from 'styled-components'

const NavbarStyled = styled(Navbar)`
  padding: 0 16px;
  box-sizing: border-box;
  background-color: ${props => props.theme.grey900};
  background-image: url('/img/bg-illo.svg');
`

const NavbarBrandStyled = styled(NavbarBrand)`
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: center;
`

const BlockchainLogoImage = styled(Image)`
  width: 240px;
  display: block;
  ${media.tablet`
    width: 200px;
  `}
`

const PublicBrand = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
`

const Header = () => (
  <>
    <NavbarStyled height='112px'>
      <NavbarBrandStyled>
        <PublicBrand>
          <Link href='https://www.blockchain.com'>
            <BlockchainLogoImage name='blockchain-logo' height='24px' />
          </Link>
        </PublicBrand>
      </NavbarBrandStyled>
    </NavbarStyled>
    <Announcements type='service' alertArea='public' />
  </>
)

export default Header
