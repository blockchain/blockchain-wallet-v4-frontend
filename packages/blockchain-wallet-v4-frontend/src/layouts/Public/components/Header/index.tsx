import React from 'react'
import { Image, Link } from 'blockchain-info-components'
import styled from 'styled-components'

import Announcements from 'components/Announcements'
import { Navbar, NavbarBrand } from 'components/Navbar'
import { media } from 'services/styles'

const NavbarStyled = styled(Navbar)`
  padding: 0 16px;
  box-sizing: border-box;
  background-color: ${props => props.theme.grey900};
  background-image: url('/img/bg-pattern.svg');
`
const NavbarBrandStyled = styled(NavbarBrand)`
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: center;
  padding-left: 0;
`

const BlockchainLogoImage = styled(Image)`
  width: 240px;
  display: block;
  ${media.tablet`
    width: 200px;
  `}
`
const PublicBrand = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
`

const Header = () => (
  <>
    <NavbarStyled height='112px'>
      <NavbarBrandStyled>
        <PublicBrand>
          <Link href='https://www.blockchain.com'>
            <BlockchainLogoImage name='blockchain-logo' height='24px' />
          </Link>
        </PublicBrand>
      </NavbarBrandStyled>
    </NavbarStyled>
    <Announcements type='service' alertArea='public' />
  </>
)

export default Header
