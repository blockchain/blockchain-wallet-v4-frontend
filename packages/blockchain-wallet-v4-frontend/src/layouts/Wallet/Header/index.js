import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { actions } from 'data'
import Header from './template.js'

class HeaderContainer extends React.PureComponent {
  render () {
    return (
      <Header
        handleToggle={() => this.props.actions.layoutWalletMenuToggleClicked()}
      />
    )
  }
}

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions.components.layoutWallet, dispatch)
})

export default connect(
  null,
  mapDispatchToProps
)(HeaderContainer)
