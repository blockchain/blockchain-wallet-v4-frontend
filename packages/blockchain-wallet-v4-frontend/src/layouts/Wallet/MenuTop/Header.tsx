import React, { useCallback } from 'react'
import { FormattedMessage } from 'react-intl'

import { Navbar } from 'components/Navbar'
import { PrimaryNavItem } from 'components/Navbar/Navbar'
import { Analytics, ModalName } from 'data/types'

import { Props } from '.'

type OwnProps = Props & {
  history: { push: (path: string) => void }
}

const Header = (props: OwnProps) => {
  const {
    analyticsActions,
    currentTier,
    featureFlags,
    history,
    invitations,
    isDexEligible,
    isKycVerificationEnabled,
    isReferralAvailable,
    isReferralEnabled,
    modalActions,
    refreshActions,
    sessionActions,
    settingsActions,
    walletDebitCardEnabled
  } = props

  const isGoldVerified = currentTier === 2

  const refreshCallback = useCallback(() => {
    refreshActions.refreshClicked()
  }, [refreshActions])

  const logoutCallback = useCallback(() => {
    sessionActions.logout()
  }, [sessionActions])

  const sendCallback = useCallback(() => {
    modalActions.showModal(ModalName.SEND_CRYPTO_MODAL, { origin: 'Header' })
  }, [modalActions])

  const receiveCallback = useCallback(() => {
    modalActions.showModal(ModalName.REQUEST_CRYPTO_MODAL, { origin: 'FeaturesTopNav' })
  }, [modalActions])

  const fabCallback = useCallback(() => {
    analyticsActions.trackEvent({
      key: Analytics.FAB_CLICKED,
      properties: {}
    })
    modalActions.showModal(ModalName.TRADE_MODAL, {
      origin: 'Header'
    })
  }, [modalActions, analyticsActions])

  const trackEventCallback = useCallback(
    (eventName) => {
      settingsActions.generalSettingsInternalRedirect(eventName)
    },
    [settingsActions]
  )

  const limitsCallback = useCallback(() => {
    modalActions.showModal(ModalName.TRADING_LIMITS_MODAL, {
      origin: 'Header'
    })
    trackEventCallback('TradingLimits')
  }, [modalActions, trackEventCallback])

  const referAFriendCallback = useCallback(() => {
    modalActions.showModal(ModalName.REFERRAL_LANDING_MODAL, {
      origin: 'Header'
    })
    trackEventCallback('Referral')
  }, [modalActions, trackEventCallback])

  const taxCenterCallback = useCallback(() => {
    history.push('/tax-center')

    analyticsActions.trackEvent({
      key: Analytics.TAX_CENTER_CLICKED,
      properties: {
        origin: 'SETTINGS'
      }
    })
  }, [analyticsActions, history])

  const primaryNavItems: PrimaryNavItem[] = [
    {
      dest: '/home',
      e2e: 'homeLink',
      text: <FormattedMessage id='copy.home' defaultMessage='Home' />
    },
    {
      dest: '/prices',
      e2e: 'pricesLink',
      text: <FormattedMessage id='copy.prices' defaultMessage='Prices' />
    },
    {
      dest: '/earn',
      e2e: 'earnLink',
      text: <FormattedMessage id='copy.earn' defaultMessage='Earn' />
    }
  ]

  if (invitations.nftBuySell) {
    primaryNavItems.push({
      dest: '/nfts/view',
      e2e: 'nftsLink',
      text: <FormattedMessage id='layouts.wallet.menuleft.navigation.nfts' defaultMessage='NFTs' />
    })
  }

  if (walletDebitCardEnabled) {
    primaryNavItems.push({
      dest: '/debit-card',
      e2e: 'debitCardLink',
      text: <FormattedMessage id='copy.card' defaultMessage='Card' />
    })
  }

  if (featureFlags.dex && isDexEligible) {
    primaryNavItems.push({
      dest: '/dex',
      e2e: 'dexLink',
      isNew: true,
      text: <FormattedMessage id='copy.dex' defaultMessage='DEX' />
    })
  }

  return (
    <Navbar
      primaryNavItems={primaryNavItems}
      fabClickHandler={fabCallback}
      isReferralAvailable={!!isReferralAvailable && isGoldVerified && isReferralEnabled}
      isReferralRetrievalEnabled={featureFlags.isReferralRetrievalEnabled}
      limitsClickHandler={limitsCallback}
      referAFriendHandler={referAFriendCallback}
      logoutClickHandler={logoutCallback}
      isKycVerificationEnabled={isKycVerificationEnabled}
      receiveClickHandler={receiveCallback}
      refreshClickHandler={refreshCallback}
      sendClickHandler={sendCallback}
      taxCenterClickHandler={taxCenterCallback}
      trackEventCallback={trackEventCallback}
    />
  )
}

export default Header
