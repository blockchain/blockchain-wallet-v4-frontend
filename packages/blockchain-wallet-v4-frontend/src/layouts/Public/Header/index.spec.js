import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from './index'

jest.mock('components/Announcements', () => ({
  ServiceAnnouncement: ''
}))
jest.mock('blockchain-info-components', () => ({ Image: '', Link: '' }))

describe('Public Header Component', () => {
  it('should match snapshot', () => {
    const component = shallow(<Header />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
