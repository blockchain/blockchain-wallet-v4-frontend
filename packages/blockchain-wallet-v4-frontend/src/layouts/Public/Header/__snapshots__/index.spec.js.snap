// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Public Header Component should match snapshot 1`] = `
<Fragment>
  <Navbar
    fluid={false}
    height="90px"
  >
    <NavbarBrand>
      <Component
        href="https://www.blockchain.com"
      >
        <Component
          height="20px"
          name="blockchain-vector"
        />
      </Component>
    </NavbarBrand>
  </Navbar>
  <[object Object]
    alertArea="public"
    type="service"
  />
</Fragment>
`;
